client/state/themes/selectors.js

```diff
@@ -44,29 +44,28 @@ export const getTheme = createSelector(
 			return null;
 		}
 
-		const theme = manager.getItem( themeId );
-		if ( siteId === 'wpcom' || siteId === 'wporg' ) {
-			return theme;
-		}
-
-		if ( ! theme ) {
-			return null;
-		}
-
-		// We're dealing with a Jetpack site. If we have theme info obtained from the
-		// WordPress.org API, merge it.
-		const wporgTheme = getTheme( state, 'wporg', themeId );
-		if ( ! wporgTheme ) {
-			return theme;
-		}
-		return {
-			...theme,
-			...pick( wporgTheme, [ 'demo_uri', 'download', 'taxonomies' ] )
-		};
+		return manager.getItem( themeId );
 	},
 	( state ) => state.themes.queries
 );
 
+export function getThemeInfoFromOriginalSource( state, siteId, themeId ) {
```

Comment:
Wondering if this could be `getCanonicalTheme`? The logic does mirror the similarly named query component.